src/datachain/lib/file.py

```diff
@@ -486,13 +490,223 @@ class ImageFile(File):
+
+        Args:
+            start_frame (int): Frame number to start reading from (default: 0).
+            end_frame (int): Frame number to stop reading at (default: None).
+            step (int): Step size for reading frames (default: 1).
+            format (str): Image format (default: 'jpg').
+
+        Returns:
+            Iterator[bytes]: Iterator of video frames.
+        """
+        from .video import video_frames
+
+        return video_frames(self, start_frame, end_frame, step, format)
+
+    def save_frames(
+        self,
+        output_dir: str,
+        start_frame: int = 0,
+        end_frame: Optional[int] = None,
+        step: int = 1,
+        format: str = "jpg",
+    ) -> "Iterator[VideoFrame]":
+        """
+        Saves video frames as image files.
+
+        Args:
+            output_dir (str): Output directory path.
+            start_frame (int): Frame number to start reading from (default: 0).
+            end_frame (int): Frame number to stop reading at (default: None).
+            step (int): Step size for reading frames (default: 1).
+            format (str): Image format (default: 'jpg').
+
+        Returns:
+            Iterator[VideoFrame]: List of video frame models.
+        """
+        from .video import save_video_frames
+
+        return save_video_frames(self, output_dir, start_frame, end_frame, step, format)
+
+    def save_fragment(
+        self,
+        start_time: float,
+        end_time: float,
+        output_file: str,
+    ) -> "VideoFragment":
+        """
+        Saves video interval as a new video file.
+
+        Args:
+            start_time (float): Start time in seconds.
+            end_time (float): End time in seconds.
+            output_file (str): Output file path.
+
+        Returns:
+            VideoFragment: Video fragment model.
+        """
+        from .video import save_video_fragment
+
+        return save_video_fragment(self, start_time, end_time, output_file)
+
+    def save_fragments(
+        self,
+        intervals: list[tuple[float, float]],
+        output_dir: str,
+    ) -> "Iterator[VideoFragment]":
+        """
+        Saves video intervals as new video files.
+
+        Args:
+            intervals (list[tuple[float, float]]): List of start and end times
+                                                   in seconds.
+            output_dir (str): Output directory path.
+
+        Returns:
+            Iterator[VideoFragment]: List of video fragment models.
+        """
+        from .video import save_video_fragments
+
+        return save_video_fragments(self, intervals, output_dir)
+
+
+class VideoFragment(VideoFile):
+    """`DataModel` for reading video fragments."""
+
+    start: float = Field(default=-1.0)
+    end: float = Field(default=-1.0)
+    orig: File
+
+
+class VideoFrame(VideoFile):
+    """`DataModel` for reading video frames."""
+
+    frame: int = Field(default=-1)
+    timestamp: float = Field(default=-1.0)
+    orig: File
```

Comment:
what is `orig`?

Reply:
I mean - do we really need it? Considering that is a subclass of a file already? Or should it be a subclass of an ImageFile? is orig then a link to the video? should it be VideoFile then? (here and above in VideoFragment)

Reply:
Since `VideoFrame` is an image file and `VideoFragment` is a video file fragment (clip), orig is a link to the video file from which these frame and fragment were taken. I took this from a @dmpetrov example.

I’m still not sure if we need this in the base library models. It looks like overhead in the general case, and if anyone needs it, they can easily add the original file either as a separate signal or by creating a custom model.

Reply:
yep, let's remove it for now for simplicity

Reply:
Removed